common/sgx/tcbinfo.c

```diff
@@ -613,88 +785,82 @@ static oe_result_t _read_tcb_info_tcb_level_v2(
 }
 
 /**
- * type = tcbInfo
- * V1 Schema:
- * {
- *    "version" : integer,
- *    "issueDate" : string,
- *    "nextUpdate" : string,
- *    "fmspc" : "hex string (12 nibbles)"
- *    "pceId" : "hex string (4 nibbles)"
- *    "tcbLevels" : [ objects of type oe_tcb_info_tcb_level_t ]
- * }
- *
- * V2 Schema:
- * {
- *    "version" : integer,
- *    "issueDate" : string,
- *    "nextUpdate" : string,
- *    "fmspc" : "hex string (12 nibbles)"
- *    "pceId" : "hex string (4 nibbles)"
- *    "tcbType" : integer
- *    "tcbEvaluationDataNumber" : integer
- *    "tcbLevels" : [ objects of type oe_tcb_info_tcb_level_t ]
- * }
+ * Read common fields among TCB Info V1, V2 and V3's tcbInfo object
+ * In the order of apperance, these fields are:
+ * - version, issueDate, nextUpdate, fmspc
+ * Fields optional for V1, but present to V2 and V3:
+ * - pceId
+ * Fields exclusive to V2 and V3:
+ * - tcbType, tcbEvaluationDataNumber
  */
-static oe_result_t _read_tcb_info(
+static oe_result_t _read_tcb_info_common_fields(
     const uint8_t* tcb_info_json,
     const uint8_t** itr,
     const uint8_t* end,
     oe_tcb_info_tcb_level_t* platform_tcb_level,
     oe_parsed_tcb_info_t* parsed_info)
 {
+    OE_UNUSED(tcb_info_json);
+    OE_UNUSED(platform_tcb_level);
+
     oe_result_t result = OE_JSON_INFO_PARSE_ERROR;
     uint64_t value = 0;
     const uint8_t* date_str = NULL;
     size_t date_size = 0;
 
-    parsed_info->tcb_info_start = *itr;
-    OE_CHECK(_read('{', itr, end));
-
     OE_TRACE_VERBOSE("Reading version");
     OE_CHECK(_read_property_name_and_colon("version", itr, end));
     OE_CHECK(_read_integer(itr, end, &value));
     parsed_info->version = (uint32_t)value;
+    if (parsed_info->version == 3)
+        parsed_info->tcb_info_v3.version = parsed_info->version;
+    else
```

Comment:
No since for TCB Info V1 and V2, it's both handled by `parsed_info->tcb_info_v2`

Reply:
Thanks for the clarification. It would be good to be explicitly clear (as a comment, name, etc) in this and anything else about what versions are handled where. In this case it would be good to mention it is handling both V1 and V2.